packages/jsii-java-runtime/project/src/main/java/software/amazon/jsii/JsiiEngine.java

```diff
@@ -362,7 +363,28 @@ private JsonNode invokeCallbackSet(final SetRequest req) {
     private JsonNode invokeCallbackMethod(final InvokeRequest req, final String cookie) {
         Object obj = this.getObject(req.getObjref());
         Method method = this.findCallbackMethod(obj.getClass(), cookie);
-        return OM.valueToTree(invokeMethod(obj, method, req.getArgs().toArray()));
+        return OM.valueToTree(invokeMethod(obj,
+                                           method,
+                                           req.getArgs()
+                                              .stream()
+                                              .map(this::fromKernel)
+                                              .toArray()));
+    }
+
+    private Object fromKernel(final Object object) {
```

Comment:
This seems like a duplicate of [`JsiiObjectMapper.treeToValue`](https://github.com/awslabs/jsii/blob/dd07831003f542ec67b6c64d369271ff0e0b1b44/packages/jsii-java-runtime/project/src/main/java/software/amazon/jsii/JsiiObjectMapper.java#L96). I am assuming .NET would have a similar method to deserialize from jsii.

I am also wondering if there are other cases (such as enums, dates) that would fail in this situation.